android/app/src/main/java/it/teamdigitale/app/italiaapp/MainActivity.java

```diff
@@ -25,28 +29,84 @@ protected String getMainComponentName() {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
-        SplashScreen.show(this, R.style.SplashScreenTheme);
-        super.onCreate(savedInstanceState);
+        if (isDeviceRooted()) {
+            super.onCreate(savedInstanceState);
+            //on rooted device show message ant stop app
+            AlertDialog alertDialog = new AlertDialog.Builder(MainActivity.this).create();
+            alertDialog.setTitle("Device rooted");
+            alertDialog.setMessage("This device is rooted, you can't use this app");
+            alertDialog.setButton(AlertDialog.BUTTON_NEUTRAL, "OK",
+                    (dialog, which) -> finish());
+            alertDialog.setCancelable(false);
+            alertDialog.show();
+        } else {
+            SplashScreen.show(this, R.style.SplashScreenTheme);
+            super.onCreate(savedInstanceState);
+        }
         // Fix the problem described here:
-        // https://stackoverflow.com/questions/48072438/java-lang-illegalstateexception-only-fullscreen-opaque-activities-can-request-o 
+        // https://stackoverflow.com/questions/48072438/java-lang-illegalstateexception-only-fullscreen-opaque-activities-can-request-o
         if (android.os.Build.VERSION.SDK_INT != Build.VERSION_CODES.O) {
             setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
         }
     }
 
     @Override
     protected ReactActivityDelegate createReactActivityDelegate() {
-        return new ReactActivityDelegate(this, getMainComponentName()) {
-            @Override
-            protected ReactRootView createRootView() {
-                return new RNGestureHandlerEnabledRootView(MainActivity.this);
-            }
-        };
+        if (isDeviceRooted()) {
+            //on rooted device not attach main component
+            return new ReactActivityDelegate(this, null);
+        } else {
+            return new ReactActivityDelegate(this, getMainComponentName()) {
+                @Override
+                protected ReactRootView createRootView() {
+                    return new RNGestureHandlerEnabledRootView(MainActivity.this);
+                }
+            };
+        }
     }
 
     @Override
     public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
-      CalendarEventsPackage.onRequestPermissionsResult(requestCode, permissions, grantResults);
-      super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        CalendarEventsPackage.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+    }
+
+    /**
+     * Check rooted device
+     */
+    private boolean isDeviceRooted() {
+        //check only once
+        if (isRootedDeviceFlag == null) {
+            isRootedDeviceFlag = (checkRootMethod1() || checkRootMethod2() || checkRootMethod3());
```

Comment:
pls add a comment to some reference documentation that explains these detection methods